src/utils/floating-ui.ts

```diff
@@ -0,0 +1,392 @@
+import {
+  computePosition,
+  Placement,
+  Strategy,
+  arrow,
+  flip,
+  shift,
+  hide,
+  offset,
+  autoPlacement,
+  autoUpdate,
+  Middleware,
+  VirtualElement
+} from "@floating-ui/dom";
+import { getElementDir } from "./dom";
+
+export type ReferenceElement = VirtualElement | Element;
+
+type UIType = "menu" | "tooltip" | "popover";
+export type OverlayPositioning = Strategy;
+
+/**
+ * Placements that change based on element direction.
+ *
+ * These variation placements will automatically flip "left"/"right" depending on LTR/RTL direction.
+ *
+ * Floating-ui has no plans to offer this functionality out of the box at this time.
+ *
+ * see: https://github.com/floating-ui/floating-ui/issues/1563 and https://github.com/floating-ui/floating-ui/discussions/1549
+ */
+type VariationPlacement = "leading-start" | "leading" | "leading-end" | "trailing-end" | "trailing" | "trailing-start";
+
+type AutoPlacement = "auto" | "auto-start" | "auto-end";
+export type LogicalPlacement = AutoPlacement | Placement | VariationPlacement;
+export type EffectivePlacement = Placement;
+
+export const placements: LogicalPlacement[] = [
+  "auto",
+  "auto-start",
+  "auto-end",
+  "top",
+  "top-start",
+  "top-end",
+  "bottom",
+  "bottom-start",
+  "bottom-end",
+  "right",
+  "right-start",
+  "right-end",
+  "left",
+  "left-start",
+  "left-end",
+  "leading-start",
+  "leading",
+  "leading-end",
+  "trailing-end",
+  "trailing",
+  "trailing-start"
+];
+
+export const effectivePlacements: EffectivePlacement[] = [
+  "top",
+  "bottom",
+  "right",
+  "left",
+  "top-start",
+  "top-end",
+  "bottom-start",
+  "bottom-end",
+  "right-start",
+  "right-end",
+  "left-start",
+  "left-end"
+];
+
+export const menuPlacements: MenuPlacement[] = ["top-start", "top", "top-end", "bottom-start", "bottom", "bottom-end"];
+
+export const menuEffectivePlacements: EffectivePlacement[] = [
+  "top-start",
+  "top",
+  "top-end",
+  "bottom-start",
+  "bottom",
+  "bottom-end"
+];
+
+export const flipPlacements: EffectivePlacement[] = [
+  "top",
+  "bottom",
+  "right",
+  "left",
+  "top-start",
+  "top-end",
+  "bottom-start",
+  "bottom-end",
+  "right-start",
+  "right-end",
+  "left-start",
+  "left-end"
+];
+
+export type MenuPlacement = Extract<
+  LogicalPlacement,
+  "top-start" | "top" | "top-end" | "bottom-start" | "bottom" | "bottom-end"
+>;
+
+export const defaultMenuPlacement: MenuPlacement = "bottom-start";
+
+export interface FloatingUIComponent {
+  /**
+   * Whether the component is opened.
+   */
+  active: boolean;
```

Comment:
I believe this should be `open` to align with the recent API consistency changes.

Reply:
Nice catch. I'll update